bundles/net.tourbook/src/net/tourbook/data/TourNutritionProduct.java

```diff
@@ -0,0 +1,385 @@
+
+   /**
+    * Unique id for the {@link TourNutritionProduct} entity
+    */
+   @Id
+   @GeneratedValue(strategy = GenerationType.IDENTITY)
+   private long                       productId                = TourDatabase.ENTITY_IS_NOT_SAVED;
+
+   /**
+    * The code identifying the product:
+    * - The bar code (https://en.wikipedia.org/wiki/Barcode)
+    * OR
+    * - A string following the below pattern if it's a user's custom product "MTCUSTOMPRODUCT-XX"
+    */
+   private String                     productCode;
+
+   @ManyToOne(optional = false)
+   private TourData                   tourData;
+
+   /**
+    * The name of the product's brand
+    */
+   private String                     brand;
+
+   /**
+    * The name of the product
+    */
+   private String                     name;
+
+   /**
+    * Calories per product.
+    * Unit: kcal
+    */
+   private int                        calories;
+
+   /**
+    * Sodium amount per product.
+    * Unit: mg
+    */
+   private int                        sodium;
+
+   /**
+    * Calories per serving.
+    * Unit: kcal
+    */
+   private int                        calories_Serving;
+
+   /**
+    * Sodium amount per serving.
+    * Unit: mg
+    */
+   private int                        sodium_Serving;
+
+   /**
+    * The quantity consumed, either in number of servings or products (see {@link #quantityType}).
+    */
+   private float                      consumedQuantity;
+
+   /**
+    * The quantity type consumed:
+    * - Servings
+    * - Number of products.
+    */
+   @Enumerated(EnumType.STRING)
+   private QuantityType               quantityType             = QuantityType.Products;
+
+   /**
+    * Indicates if the nutrition product is a beverage itself or used in a
+    * beverage.c
+    */
+   private boolean                    isBeverage;
+
+   /**
+    * If the product is a beverage, the amount of liquid per product.
+    * Unit: mL
+    */
+   private int                        beverageQuantity;
+
+   /**
+    * If the product is a beverage, the amount of liquid per serving.
+    * Unit: mL
+    */
+   private int                        beverageQuantity_Serving;
+
+   @ManyToOne
+   private TourBeverageContainer      tourBeverageContainer;
+
+   /**
+    * The total number of beverage containers consumed
+    */
+   private float                      containersConsumed;
+
+   /**
+    * Unique id for manually created products because the {@link #productCode}
+    * is 0 when the product is not persisted
+    */
+   @Transient
+   private long                       _createId                = 0;
+
+   public TourNutritionProduct() {}
+
+   public TourNutritionProduct(final TourData tourData, final boolean isCustomProduct) {
+
+      this.tourData = tourData;
+      _createId = _createCounter.incrementAndGet();
+      productCode = isCustomProduct ? CUSTOMPRODUCTCODE_PREFIX + _createId : UI.EMPTY_STRING;
```

Comment:
Firstly I saw that this is not a good solution by adding some string into another field to identify a custom product. But then I also saw the weakness of this code: this can create the same `productCode` after restarting MT

I would recommend to add a new field e.g. `boolean isCustomProduct`

![image](https://github.com/mytourbook/mytourbook/assets/1283445/827b73e5-1b4f-4b01-8015-d64ad12446a1)

Reply:
Good suggestion, implemented